doc/partitioned-db.md

```diff
@@ -0,0 +1,203 @@
+# Partitioned Database
+
+When creating a Cloudant database, you can configure it as non-partitioned or partitioned.
+A partitioned database offers significant performance and cost advantages but requires you to specify a logical partitioning of your data.
```

Comment:
nitpick: instead of "but....", maybe we can just start a new sentence with "This requires you to specify..."?